test/unit/range/decorator/gap_decorator_anchor_set_test.cpp

```diff
@@ -0,0 +1,216 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/decorator/gap_decorator_anchor_set.hpp>
+#include <seqan3/std/view/subrange.hpp>
+#include <seqan3/std/view/view_all.hpp>
+#include <seqan3/std/ranges>
+
+#include "../../alignment/aligned_sequence_test_template.hpp"
+
+using namespace seqan3;
+
+template <typename inner_type_>
+class aligned_sequence<gap_decorator_anchor_set<inner_type_>> : public ::testing::Test
+{
+public:
+    // Initialiser function is needed for the typed test because the gapped_decorator
+    // will be initialised differently than the naive vector<gapped<dna>>.
+    void initialise_typed_test_container(gap_decorator_anchor_set<inner_type_> & container, dna4_vector const & target)
+    {
+        container = target;
+    }
+};
+
+template <typename t>
+class gap_decorator_anchor_set_f : public ::testing::Test {};
+
+const std::vector<dna4> dummy_obj{}; // dummy lvalue for type declaration of views
```

Comment:
This currently doesn't work with `bitcompressed_vector` (because of some bug) but should, right?